File: lib/display/ChatGroup.ts
import { Avatar, AxisAlignmentDirectional, ChatBubbleType } from "../widgets";

export interface ChatGroupProps {
    children: any[],
    alignment: AxisAlignmentDirectional,
    color: string,
    type: ChatBubbleType,
    borderRadius?: any,
    avatarPrefix?: any
}

export function ChatGroup(id: string, {
    children = [],
    alignment = AxisAlignmentDirectional.start,
    color = "#7700ff",
    type = ChatBubbleType.sharpCorner,
    borderRadius = 10,
    avatarPrefix = Avatar('#avatar', { initials: "AR" })
}: ChatGroupProps){
    return {
        id,
        name: "ChatGroup",
        props: {
            children,
            alignment,
            color,
            type,
            borderRadius,
            avatarPrefix
        }
    }
}

File: flutter_render/lib/src/dispaly/ChatGroup.dart
import 'package:flutter/widgets.dart';
import 'package:neneui_render/src/enum.dart';
import 'package:neneui_render/src/parser/Color.dart';
import 'package:neneui_render/src/parser/Core.dart';
import 'package:shadcn_flutter/shadcn_flutter_experimental.dart';

class DChatGroup {
  static Widget run({
    required BuildContext context,
    required Map<String, dynamic> data,
    required Function reRender,
    required Function event,
  }) {
    if (data['name'] == "ChatGroup") {
      event(Events.REGISTER_ID, {'id': data['id'], 'props': data['props']});

      return ChatGroup(
        alignment: CoreParser.parseAxisAlignDirect(data['props']['alignment']),
        color: ColorParse.parseColor(data['props']['color']),
        type: CoreParser.parseChaBubbleType(data['props']['type']),
        avatarPrefix: reRender(data['props']['avatarPrefix']),
        children: [
          for (var ui in List.from(data['props']['children'])) reRender(ui),
        ],
      );
    } else {
      return SizedBox();
    }
  }
}
